src/Kopernicus/Components/HazardousBody.cs

```diff
@@ -77,10 +78,10 @@ private void Start()
         }
 
         /// <summary>
-        /// Update the heat
+        /// Update the heat. Heating is physics phenomenon so do it in the physics loop.
         /// </summary>
         /// <returns></returns>
-        private void Update()
+        private void FixedUpdate()
```

Comment:
FixedUpdate is not that fixed. It's a physics engine update, so it gets called only when the game is running physics. When the game is paused, physics engine is also paused. On time warp there is no physics either. On physics warp, it will run but the TimeWarp.fixedDeltaTime will be greater. Also when the physics engine is lagging, it will run at a slower rate in real time but constant rate in physics time.